core/signeddata.go

```diff
@@ -807,6 +807,72 @@ func (s *BeaconCommitteeSelection) UnmarshalJSON(input []byte) error {
 	return s.BeaconCommitteeSelection.UnmarshalJSON(input)
 }
 
+// SyncCommitteeSelection wraps an eth2exp.SyncCommitteeSelection and implements SignedData.
+type SyncCommitteeSelection struct {
+	eth2exp.SyncCommitteeSelection
+}
+
+func (s SyncCommitteeSelection) MessageRoot() ([32]byte, error) {
+	data := altair.SyncAggregatorSelectionData{
+		Slot:              s.Slot,
+		SubcommitteeIndex: uint64(s.SubcommitteeIndex),
+	}
+
+	return data.HashTreeRoot()
```

Comment:
this can be done inline:
```
        return altair.SyncAggregatorSelectionData{
		Slot:              s.Slot,
		SubcommitteeIndex: uint64(s.SubcommitteeIndex),
	}.HashTreeRoot()
```